lib/src/checkbox/Checkbox.tsx

```diff
@@ -160,26 +163,95 @@ const getNotDisabledColor = (props, element) => {
       return props.backgroundType && props.backgroundType === "dark"
         ? props.theme.backgroundColorCheckedOnDark
         : props.theme.backgroundColorChecked;
+    case "hoverBackground":
+      return props.backgroundType && props.backgroundType === "dark"
+        ? props.theme.hoverBackgroundColorCheckedOnDark
+        : props.theme.hoverBackgroundColorChecked;
     case "border":
       return props.backgroundType && props.backgroundType === "dark"
         ? props.theme.borderColorOnDark
         : props.theme.borderColor;
+    case "hoverBorder":
+      return props.backgroundType && props.backgroundType === "dark"
+        ? props.theme.hoverBorderColorOnDark
+        : props.theme.hoverBorderColor;
     case "label":
       return props.backgroundType && props.backgroundType === "dark"
         ? props.theme.fontColorOnDark
         : props.theme.fontColor;
   }
 };
 
-const LabelContainer = styled.span`
+type LabelContainerPropsType = {
+  disabled: boolean;
+  backgroundType: "dark" | "light";
+};
+const LabelContainer = styled.span<LabelContainerPropsType>`
   color: ${(props) => (props.disabled ? getDisabledColor(props, "label") : getNotDisabledColor(props, "label"))};
   font-family: ${(props) => props.theme.fontFamily};
   font-size: ${(props) => props.theme.fontSize};
   font-weight: ${(props) => props.theme.fontWeight};
   cursor: ${(props) => (props.disabled ? "not-allowed" : "pointer")};
 `;
 
+const ValueInput = styled.input`
+  display: none;
+`;
+
 const CheckboxContainer = styled.span`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  height: 24px;
+  width: 24px;
+`;
+
+type CheckboxInputPropsType = {
+  backgroundType: "dark" | "light";
+  checked: boolean;
+  disabled: boolean;
+};
+const Checkbox = styled.span<CheckboxInputPropsType>`
+  box-sizing: border-box;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  height: 18px;
+  width: 18px;
+  border-radius: 2px;
+  border: solid 2px
+    ${(props) => (props.disabled ? getDisabledColor(props, "border") : getNotDisabledColor(props, "border"))};
+  background-color: ${(props) =>
+    props.checked
+      ? props.disabled
+        ? getDisabledColor(props, "check")
+        : getNotDisabledColor(props, "check")
+      : "transparent"};
+  color: ${(props) =>
+    props.disabled ? getDisabledColor(props, "background") : getNotDisabledColor(props, "background")};
+
+  &:focus-visible {
```

Comment:
Should we update this to `:focus, :focus-visible` selector? 🤔

Reply:
I think we should change it to :focus to make it consistent with our other components.